src/coreclr/src/inc/corinfoinstructionset.h

```diff
@@ -50,15 +50,16 @@ enum CORINFO_InstructionSet
     InstructionSet_POPCNT=16,
     InstructionSet_Vector128=17,
     InstructionSet_Vector256=18,
-    InstructionSet_X86Base_X64=19,
-    InstructionSet_BMI1_X64=20,
-    InstructionSet_BMI2_X64=21,
-    InstructionSet_LZCNT_X64=22,
-    InstructionSet_POPCNT_X64=23,
-    InstructionSet_SSE_X64=24,
-    InstructionSet_SSE2_X64=25,
-    InstructionSet_SSE41_X64=26,
-    InstructionSet_SSE42_X64=27,
+    InstructionSet_AVX512F=19,
```

Comment:
This also needs JIT/EE interface GUID update because of is revving JIT/VM contract. It has other side-effects that are not ideal e.g. AVX512F will be recognized as instruction set on the crossgen command line, even though crossgen won't ever do anything with it.

I think it would be a lot simpler and maintainable to just do a local detection in GC. It is really only needed for Windows - we can just use the built-in function multi-versioning for Unix.

Reply:
Ok, I implemented local detection in the GC instead and undid the changes on the VM side.

Reply:
Thanks!